src/main/java/com/google/sps/servlets/WeatherServlet.java

```diff
@@ -0,0 +1,62 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.inject.Inject;
+
+import com.google.gson.Gson;
+import com.google.maps.model.LatLng;
+import com.google.sps.data.Weather;
+import com.google.sps.data.GeoCoding;
+
+@WebServlet("/weather")
+public class WeatherServlet extends HttpServlet {
+  Weather WeatherObject;
+  GeoCoding GeoCodingObject;
+
+  @Inject
+  WeatherServlet(Weather WeatherObject, GeoCoding GeoCodingObject) {
+    this.WeatherObject = WeatherObject;
+    this.GeoCodingObject = GeoCodingObject;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    String location = request.getParameter("location");
+    Integer hours = parseIntegerFromString(request.getParameter("hours"));
+    Integer days = parseIntegerFromString(request.getParameter("days"));
+    LatLng latlng = GeoCodingObject.fromAddressToLatLng(location);
```

Comment:
Maybe check better is location is null so you can throw an error from the servlet to the front end for bad request